Objects/longobject.c

```diff
@@ -3695,6 +3695,10 @@ static PyObject *
 long_div(PyObject *a, PyObject *b)
 {
     PyLongObject *div;
+    if (b == _PyLong_One && PyLong_CheckExact(a)) {
```

Comment:
This branch *may* impact the normal path, we certainly should benchmark this with PGO and cpu isolation